prow/tide/history/history.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package history
+
+import (
+	"encoding/json"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+// FE should support:
+// - filter by repo (?branch) (?pr)
+// - filter by author
+// - filter by action
+// - filter by errored state
+// - filter 'before time t'
+// - no-filter. i.e. all actions tide has taken in order.
+
+//
+// ???	Should `History` use a single `*recordLog` instead of per pool?
```

Comment:
I think using multiple record logs is probably best. We could make `logSizeLimit` optionally configurable per repo to give busy pools more history, but that's possibly overkill